src/main/java/com/minecolonies/api/util/WorldUtil.java

```diff
@@ -67,7 +68,7 @@ public static boolean isChunkLoaded(final LevelAccessor world, final int x, fina
             final ChunkHolder holder = ((ServerChunkCache) world.getChunkSource()).chunkMap.getVisibleChunkIfPresent(ChunkPos.asLong(x, z));
             if (holder != null)
             {
-                return holder.getFullChunkFuture().getNow(ChunkHolder.UNLOADED_LEVEL_CHUNK).isSuccess();
+                return holder.getFullStatus().isOrAfter(FullChunkStatus.FULL);
```

Comment:
This actually just tells us if it will be loaded eventually, not if it does exist right now. Why not use the "holder.getChunkIfPresent" check for both?

Reply:
Not sure how I missed that, but getChunkIfPresent will work here. Thanks for the review, PR is updated.

Reply:
After talking with @Spottedleaf we determined that both the full status and getChunkIfPresent should be used due to some implementation details of the Moonrise chunk system, for vanilla it is redundant but doesn't hurt anything.
For ChunkCache, getChunkIfPresent is fine on its own as entity ticking is already checked which means we already know the full status will be FULL or after.

Changing this behavior to closer align with vanilla is something I'm interested in but not sure about a timeline for. If/when we make that change the code in this PR would still be valid, as it is for vanilla.